### ISSUE TITLE
Support calling matches, after and evaluate functions in axe.configure()

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2188]
To improve configurability of axe-core, we want to allow mapping new checks / rules to existing matches, evaluate and after functions. For example:

```js
axe.configure({
  rules: [{
    "id": "custom-colors",
    "matches": "color-contrast-matches.js",
    "any": ["custom-colors"],
    ...more
  }],
  checks: [{
    "id": "custom-colors",
    "evaluate": "color-contrast-evaluate.js",
    ...more
  }]
})
```

Doing this makes it much easier for custom rule authors to reuse existing functionality available in axe.  We'll need to create a map of all matches / evaluate / after functions that we can do a lookup from.

**Comment by straker** [comment]
This work has been completed in the ES module migration work. So doing this will now work:

```js
axe.configure({
  rules: [{
    id: 'custom-rule',
    all: ['custom-check']
  }],
  checks: [{
    id: 'custom-check',
    evaluate: 'region-evaluate'
  }]
});
```

**Comment by padmavemulapati** [comment]
Verified with latest axe-core code based-
using the configure script ' axe.configure({
  rules: [{
    id: 'custom-rule',
    all: ['custom-check']
  }],
  checks: [{
    id: 'custom-check',
    evaluate: 'region-evaluate'
  }]
}); `  , can verify that it doesn't fail. Then  find a page (region rule - it was failing, the custom-rule should fail as well with the same node),  that has a violation and create a custom rule/check that uses the failed check and verity it gives the same violation results . 
Below are the screenshots.
![image (27)](https://user-images.githubusercontent.com/47409579/82138884-6996ad00-9841-11ea-80ef-f74016c5acfa.png)
![image (28)](https://user-images.githubusercontent.com/47409579/82138887-6e5b6100-9841-11ea-9149-e2509078c023.png)

**Comment by jeankaplansky** [comment]
@straker: Any customer-facing docs in this?

**Comment by straker** [comment]
Ya, there's going to be a whole lot of new docs for this. This is part of the custom rules overhaul for 4.0. Probably best to talk with Wilco on our current plans and direction for 4.0.

**Comment by jeankaplansky** [comment]
@WilcoFiers, @erik-deque: Who's writing the docs for axe-core 4.0? Is this task staying with Steve?

**Comment by jeankaplansky** [comment]
included in internal release notes: 7/20/2020
